csharp/src/Ice/Protocol.cs

```diff
@@ -158,6 +159,45 @@ public static Ice.OutputStream CreateFailureResponseFrame(System.Exception excep
             return ostr;
         }
 
+        internal static Ice.Current CreateCurrent(int requestId, Ice.InputStream requestFrame,
+            Ice.ObjectAdapter adapter, Ice.Connection? connection = null)
+        {
+            int start = requestFrame.Pos;
+            var identity = new Ice.Identity(requestFrame);
+
+            // For compatibility with the old FacetPath.
+            string[] facetPath = requestFrame.ReadStringArray();
+            string facet;
+            if (facetPath.Length > 0)
+            {
+                if (facetPath.Length > 1)
+                {
+                    throw new Ice.MarshalException();
+                }
+                facet = facetPath[0];
+            }
+            else
+            {
+                facet = "";
+            }
+
+            string operation = requestFrame.ReadString();
+            byte mode = requestFrame.ReadByte();
+            var context = new Dictionary<string, string>();
+            int sz = requestFrame.ReadSize();
```

Comment:
Shouldn't this code use `var context = requestFrame.ReadContext()`

Reply:
It should, yes (I didn't rewrite it, just moved it there). It's possible we could ever use a helper Slice struct here.

Reply:
Fixed!